crates/solidity/outputs/rust/lib/src/generated/cst.rs

```diff
@@ -1,246 +1,133 @@
 // This file is generated automatically by infrastructure scripts. Please don't edit by hand.
 
-use super::kinds::{RuleKind, TokenKind};
-use super::lex;
-use serde::Serialize;
+use std::ops::Range;
 use std::rc::Rc;
+
+use serde::Serialize;
+
+use super::kinds::*;
+
 #[derive(Clone, Debug, PartialEq, Eq, Serialize)]
 pub enum Node {
     Rule {
         kind: RuleKind,
+        span: Range<usize>,
         children: Vec<Rc<Node>>,
     },
     Token {
         kind: TokenKind,
-        lex_node: Rc<lex::Node>,
+        span: Range<usize>,
         #[serde(skip_serializing_if = "Vec::is_empty")]
         trivia: Vec<Rc<Node>>,
     },
-    #[doc = r" For anonymous groups referenced from AST nodes i.e. `delimited_by`"]
-    Group { children: Vec<Rc<Node>> },
 }
+
 impl Node {
-    pub fn rule(kind: RuleKind, children: Vec<Option<Rc<Self>>>) -> Option<Rc<Self>> {
-        let children: Vec<_> = children.into_iter().filter_map(|e| e).collect();
-        if children.is_empty() {
-            return None;
+    pub fn is_empty(&self) -> bool {
+        match self {
+            Self::Rule { children, .. } => children.is_empty(),
+            Self::Token { trivia, span, .. } => span.is_empty() && trivia.is_empty(),
```

Comment:
Can `Node::Token` ever be empty? not sure what is the use case for that.

Reply:
code deleted